modules/Bio/EnsEMBL/Compara/RunnableDB/DataCheckFan.pm

```diff
@@ -36,11 +36,17 @@ use base ('Bio::EnsEMBL::DataCheck::Pipeline::DataCheckFan', 'Bio::EnsEMBL::Comp
 
 sub fetch_input {
     my $self = shift;
-    $self->param('dba', $self->compara_dba);
-
-    my $prev_dba = Bio::EnsEMBL::Compara::DBSQL::DBAdaptor->go_figure_compara_dba( 'compara_prev' );
-    $self->param('old_server_uri', $prev_dba->url);
 
+    $self->param('dba', $self->compara_dba);
+    # The pipeline may not be in the registry_file so server_uri needs to be explicitly passed
+    unless ( $self->param('registry_file') ) {
+        $self->param('server_uri', $self->param('compara_db'));
+    }
+    # For some pipelines the previous db is irrelevant so default to same db
```

Comment:
I like this suggestion/comment, but maybe it would be better to add it to the DESCRIPTION of the runnable? I feel it is not exactly related with what the code below is doing.